### ISSUE TITLE
Apollo Server and Client example doesn't work correctly

### ISSUE THREAD

**Issue by Industrial** [zeit/next.js#9465]
# Bug report

## Describe the bug

1. This line exposes Apollo Server on the `/api/graphql` route: https://github.com/zeit/next.js/blob/canary/examples/api-routes-apollo-server-and-client/pages/api/graphql.js#L12
2. This line causes `apollo-link-schema` to be used on SSR, effectively bundling all GraphQL resolvers into the server side bundle: https://github.com/zeit/next.js/blob/canary/examples/api-routes-apollo-server-and-client/apollo/client.js#L139
3. This line returns an object in the GraphQL Resolver: https://github.com/zeit/next.js/blob/canary/examples/api-routes-apollo-server-and-client/apollo/resolvers.js#L4
  3.1 If you perform actual work inside the resolver where you use a module like `fs` or `mongodb` or `firebase-admin` to implement the resolver, it will give you a Webpack bundling error that it `cannot resolve module 'fs'`. This is due to step 2.

**Comment by Industrial** [comment]
I made a project that tries to merge several examples:
1. I'm running/deploying on Firebase so I started with https://github.com/zeit/next.js/tree/canary/examples/with-firebase-hosting-and-typescript.
1. Next to a React app with SSR with Next.js I want to deploy an Apollo Server and Client so I went with https://github.com/zeit/next.js/tree/canary/examples/api-routes-apollo-server-and-client.

This is the error I'm getting:

```
$ next dev src/app
[ wait ]  starting the development server ...
[ info ]  waiting on http://localhost:3000 ...
[ info ]  bundled successfully, waiting for typecheck results...
[ wait ]  compiling ...
[ info ]  bundled successfully, waiting for typecheck results...
[ event ] build page: /assets
[ wait ]  compiling ...
[ error ] /Users/tom/Code/Code9/SmallCrypto/small-crypto-next/node_modules/google-auth-library/build/src/auth/googleauth.js
Module not found: Can't resolve 'child_process' in '/Users/tom/Code/Code9/SmallCrypto/small-crypto-next/node_modules/google-auth-library/build/src/auth'
```

This is a hard to reproduce error because you need a Firebase Project and download the `serviceAccountKey.json` to use with `firebase-admin` and place it in `src/functions/keys` in order to get the firebase client to work. This is something I can't share or put into version control :-)

**Comment by Industrial** [comment]
@Timer Hi!

This is my project: https://github.com/Industrial/small-crypto-next

This is the Apollo Client: https://github.com/Industrial/small-crypto-next/blob/master/src/app/apollo/client.tsx which comes from this example https://github.com/zeit/next.js/blob/canary/examples/api-routes-apollo-server-and-client/apollo/client.js

At this line it needs to load the GraphQL Schema on the Server Side Rendering: https://github.com/Industrial/small-crypto-next/blob/master/src/app/apollo/client.tsx#L23

I have a "test" file here: https://github.com/Industrial/small-crypto-next/blob/master/src/app/apollo/resolvers/query/foo.json
In this resolver I am using `fs.readFileSync` to open that file. This is a server only operation and should not be bundled in the client bundle.

The error I am getting is:
```
Module not found: Can't resolve 'fs' in '/Users/tom/Code/Code9/SmallCrypto/small-crypto-next/src/app/apollo/resolvers/query'
ModuleNotFoundError: Module not found: Error: Can't resolve 'fs' in '/Users/tom/Code/Code9/SmallCrypto/small-crypto-next/src/app/apollo/resolvers/query'
    at factory.create (/Users/tom/Code/Code9/SmallCrypto/small-crypto-next/node_modules/webpack/lib/Compilation.js:888:10)
    at factory (/Users/tom/Code/Code9/SmallCrypto/small-crypto-next/node_modules/webpack/lib/NormalModuleFactory.js:401:22)
    at resolver (/Users/tom/Code/Code9/SmallCrypto/small-crypto-next/node_modules/webpack/lib/NormalModuleFactory.js:130:21)
    at asyncLib.parallel (/Users/tom/Code/Code9/SmallCrypto/small-crypto-next/node_modules/webpack/lib/NormalModuleFactory.js:224:22)
    at /Users/tom/Code/Code9/SmallCrypto/small-crypto-next/node_modules/neo-async/async.js:2830:7
    at /Users/tom/Code/Code9/SmallCrypto/small-crypto-next/node_modules/neo-async/async.js:6877:13
    at normalResolver.resolve (/Users/tom/Code/Code9/SmallCrypto/small-crypto-next/node_modules/webpack/lib/NormalModuleFactory.js:214:25)
    at doResolve (/Users/tom/Code/Code9/SmallCrypto/small-crypto-next/node_modules/enhanced-resolve/lib/Resolver.js:213:14)
    at hook.callAsync (/Users/tom/Code/Code9/SmallCrypto/small-crypto-next/node_modules/enhanced-resolve/lib/Resolver.js:285:5)
    at _fn0 (eval at create (/Users/tom/Code/Code9/SmallCrypto/small-crypto-next/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:15:1)
    at resolver.doResolve (/Users/tom/Code/Code9/SmallCrypto/small-crypto-next/node_modules/enhanced-resolve/lib/UnsafeCachePlugin.js:44:7)
    at hook.callAsync (/Users/tom/Code/Code9/SmallCrypto/small-crypto-next/node_modules/enhanced-resolve/lib/Resolver.js:285:5)
    at _fn0 (eval at create (/Users/tom/Code/Code9/SmallCrypto/small-crypto-next/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:15:1)
    at hook.callAsync (/Users/tom/Code/Code9/SmallCrypto/small-crypto-next/node_modules/enhanced-resolve/lib/Resolver.js:285:5)
    at _fn0 (eval at create (/Users/tom/Code/Code9/SmallCrypto/small-crypto-next/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:27:1)
    at resolver.doResolve (/Users/tom/Code/Code9/SmallCrypto/small-crypto-next/node_modules/enhanced-resolve/lib/DescriptionFilePlugin.js:67:43)
```

It seems like Webpack is bundling my GraphQL Resolver in the client bundle. I don't know why.

To reproduce:

`git clone https://github.com/Industrial/small-crypto-next`
`cd small-crypto-next`
`yarn`
`yarn dev`
`open http://localhost:3000/assets`

**Comment by Industrial** [comment]
Apparently you have to use `typeof window === 'undefined'` directly in your code to bundle it on the server only and you can't use `function isServer() { return typeof window === 'undefined' }` because then Nextjs/Webpack will bundle it in the client.

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
